.goreleaser.yaml

```diff
@@ -1,27 +1,67 @@
-# This is an example .goreleaser.yml file with some sensible defaults.
-# Make sure to check the documentation at https://goreleaser.com
 before:
   hooks:
     - go mod tidy
 builds:
-  - main: ./cmds/ocm/main.go
-    env:
-      - CGO_ENABLED=0
+  - <<: &build_defaults
+      binary: ocm
+      main: ./cmds/ocm/main.go
+      ldflags:
+        - -s -w -X github.com/open-component-model/ocm/pkg/version.gitVersion={{.Version}} -X github.com/open-component-model/ocm/pkg/version.gitCommit={{.Commit}} -X github.com/open-component-model/ocm/pkg/version.buildDate={{.CommitDate}}
+      env:
+        - CGO_ENABLED=0
+    id: linux
     goos:
       - linux
-      - windows
+    goarch:
+      - amd64
+      - arm64
+      - arm
+    goarm:
+      - 7
+  - <<: *build_defaults
+    id: darwin
+    goos:
       - darwin
+    goarch:
+      - amd64
+      - arm64
+  - <<: *build_defaults
+    id: windows
+    goos:
+      - windows
 archives:
-  - name_template: >-
-      {{ .ProjectName }}_
-      {{- title .Os }}_
-      {{- if eq .Arch "amd64" }}x86_64
-      {{- else if eq .Arch "386" }}i386
-      {{- else }}{{ .Arch }}{{ end }}
-checksum:
-  name_template: 'checksums.txt'
-snapshot:
-  name_template: "{{ incpatch .Version }}-next"
+  - name_template: "{{ .Binary }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}"
```

Comment:
Yes there was a "Title" function being used to transform the Os value..

Reply:
The original release didn't include the version in the archive name. If the only consumer we have is the github/action then I propose that I can update that in a follow on PR.

I have done this also for the setup.sh script which is used to install ocm via `curl | bash`.

Reply:
That is fine, I think we don't need to update the old releases accordingly, because they will never be used anymore by the action.

We should stick to notations used in the go ecosystem, for example https://go.dev/dl/. They add the version and use directly the GOARCH and GOOS names and dashed (`-`).

So ocm-<version>-linux-amd64.tgz or ocm-<version>.linux-amd64,tgz (according to the Go DL) should be fine.
Could provide matching PRs for the ocm ocm-setup-action and ocm-action project?

Reply:
> We should stick to notations used in the go ecosystem, for example https://go.dev/dl/. They add the version and use directly the GOARCH and GOOS names and dashed (-).

Agreed.

> Could you provide matching PRs for the ocm ocm-setup-action and ocm-action project?

Yes, will do.